### DISCUSSION TITLE
ignore warnings in `tar_make` and friends

### DISCUSSION THREAD

**Discussion by jennysjaarda** [ropensci/targets#703] [Help] [answered]
Hello, 
I am making a brief presentation of the `targets` package for my group (there is a lot of interest!) and I am getting warnings regarding package version being incompatible with R versions. It's nothing to be concerned about I reckon, but I want to remove these message from the presentation somehow. 

```
library(targets)

tar_script({
    library(targets)
    options(crayon.enabled = FALSE)
    options(tidyverse.quiet = TRUE)
    tar_option_set(packages = c("tidyverse"))
    list(
        tar_target(x, 1:10)
    )
})


tar_make()
#> • start target x
#> • built target x
#> • end pipeline
#> Warning messages:
#> 1: package ‘ggplot2’ was built under R version 4.1.1 
#> 2: package ‘tidyr’ was built under R version 4.1.1 
#> 3: package ‘readr’ was built under R version 4.1.1 
#> 4: package ‘stringr’ was built under R version 4.1.1 
#> 5: 1 targets produced warnings. Run tar_meta(fields = warnings) for the messages.
```
Is there a simple work around for this? adding `suppressWarnings` to `tar_make()` doesn't seem to help.

**Comment by wlandau** [comment, marked as answer]
Thanks for spreading the word. As long as you are using `tar_make()` and not `tar_make_clustermq()` or `tar_make_future()`, you could call `suppressWarnings(library(tidyverse))` at the very top. Then the warnings should not come up later on. For a presentation, you could disguise that in a `source("R/functions.R")` script. 

Or you could use [`tarchetypes::tar_hook_outer()`](https://docs.ropensci.org/tarchetypes/reference/tar_hook_outer.html)

```{r}
tar_script({
    library(targets)
    options(crayon.enabled = FALSE)
    options(tidyverse.quiet = TRUE)
    tar_option_set(packages = c("tidyverse"))
    tarchetypes::tar_hook_outer(
      list(
        tar_target(x, 1:10)
      ),
      hook = suppressWarnings(.x)
    )
})

**Reply by jennysjaarda** [reply]
thanks very much!
